File: Euclidean's Algorithm/Extended/gcd_with_factors.c
#include <stdio.h>
#include <conio.h>

int findGcdAndFactors(int a,int b,int *x,int *y){
  if (a == 0) {
    *x = 0;
    *y = 1;
    return b;
  }
  int x1,y1;
  int gcd = findGcdAndFactors(b%a,b,&x1,&y1);
  *x = y1 - (b/a) * x1;
  *y = x1;
  return gcd;
}

int main() {
  int x,y,a,b,gcd;
  printf("Enter numbers\n");
  scanf("%d%d",&a,&b);
  if(a > b) {
    gcd = findGcdAndFactors(a,b,&x,&y);
  } else {
    gcd = findGcdAndFactors(b,a,&y,&x);
  }
  printf("%d\n%d\n%d",gcd,x,y);
  return 0;
}

File: Euclidean's Algorithm/Extended/gcd_with_factors.py
x,y = 0,0
def findGcdAndFactors(a,b,m,n,arr):
    global x,y
    if a == 0:
        m = 0
        n = 1
        arr.append(m)
        arr.append(n)
        arr.append(b)
        return arr
    m,n,b = arr[0],arr[1],arr[2]
    arr = []
    gcd = findGcdAndFactors(b%a,b,m,n,arr)
    x = gcd[1] - (gcd[2]//a)*gcd[0]
    y = gcd[0]
    return gcd

def main():
    global x,y
    print("Enter two number seperated by space.")
    a,b = map(int,input().split(" "))
    gcd = [x,y,b]
    gcd = findGcdAndFactors(a,b,x,y,gcd)
    print(gcd[2],gcd[0],gcd[1])

if __name__ == '__main__':
    main()
